src/utils/update/psc.to.beneficial.owner.type.mapper.ts

```diff
@@ -0,0 +1,191 @@
+    usual_residential_address: address,
+    service_address: service_address,
+    start_date: psc.notifiedOn as any as InputDate, // notified is type string not input date
+    // is_on_sanctions_list: psc.isSanctioned === "true" ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, false);
+  return result;
+};
+
+export const mapPscToBeneficialOwnerOther = (psc: CompanyPersonWithSignificantControl): BeneficialOwnerOther => {
+  const service_address = mapAddress(psc.address);
+  const principal_address = mapAddress(psc.address);
+
+  const result: BeneficialOwnerOther = {
+    id: psc.links?.self,
+    link: {
+      self: psc.links?.self
+    },
+    name: psc.name,
+    principal_address: principal_address,
+    service_address: service_address,
+    is_service_address_same_as_principal_address: isSameAddress(service_address, principal_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    legal_form: psc.identification?.legalForm,
+    law_governed: psc.identification?.legalAuthority,
+    public_register_name: psc.identification?.placeRegistered,
+    registration_number: psc.identification?.registrationNumber,
+    is_on_register_in_country_formed_in: undefined,
+    start_date: psc.notifiedOn as any as InputDate, // notified is type string not input date
+    // is_on_sanctions_list: psc.isSanctioned === "true" ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, false);
+  return result;
+
+};
+
+export const mapPscToBeneficialOwnerGov = (psc: CompanyPersonWithSignificantControl): BeneficialOwnerGov => {
+  const service_address = mapAddress(psc.address);
+  const principal_address = mapAddress(psc.address);
+
+  const result: BeneficialOwnerGov = {
+    id: psc.links?.self,
+    link: {
+      self: psc.links?.self
+    },
+    name: psc.name,
+    principal_address: principal_address,
+    service_address: service_address,
+    is_service_address_same_as_principal_address: isSameAddress(service_address, principal_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    legal_form: psc.identification?.legalForm,
+    law_governed: psc.identification?.legalAuthority,
+    start_date: psc.notifiedOn as any as InputDate, // notified is type string not input date
+    // is_on_sanctions_list: psc.isSanctioned === "true" ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, true);
+  return result;
+};
+
+const mapNatureOfControl = (psc: CompanyPersonWithSignificantControl, beneficialOwner: BeneficialOwnerIndividual| BeneficialOwnerOther, isBeneficialGov: boolean) => {
+  psc.naturesOfControl?.forEach(natureType => {
+    const controlKind = natureTypeMap.get(natureType);
+
+    if (beneficialOwner){
+      switch (controlKind) {
+          case 'BoNatureOfControl':
+            beneficialOwner.beneficial_owner_nature_of_control_types = beneficialOwnerMap.get(natureType) as unknown as NatureOfControlType[];
```

Comment:
Can use [] instead of get. Don't need two as (remove the "as unknown") I think.

Reply:
Was force to use get() as [] wasn't returning any match as I would naturally have expected